content/hacking-atom/sections/hacking-on-atom-core.md

```diff
@@ -0,0 +1,307 @@
+
+then restart Atom.  If Atom now works fine, you can make this setting permanent:
+
+``` command-line
+$ echo 32768 | sudo tee -a /proc/sys/fs/inotify/max_user_watches
+```
+
+See also [#2082](https://github.com/atom/atom/issues/2082).
+
+###### /usr/bin/env: node: No such file or directory
+
+If you get this notice when attempting to run any script, you either do not have
+Node.js installed, or node isn't identified as Node.js on your machine. If it's
+the latter, this might be caused by installing Node.js via the distro package
+manager and not nvm, so entering `sudo ln -s /usr/bin/nodejs /usr/bin/node` into
+your terminal may fix the issue. On some variants (mostly Debian based distros)
+you can use `update-alternatives` too:
+
+``` command-line
+$ sudo update-alternatives --install /usr/bin/node node /usr/bin/nodejs 1 --slave /usr/bin/js js /usr/bin/nodejs
+```
+
+###### AttributeError: 'module' object has no attribute 'script_main'
+
+If you get following error with a big traceback while building Atom:
+
+```
+sys.exit(gyp.script_main()) AttributeError: 'module' object has no attribute 'script_main' gyp ERR!
+```
+
+you need to uninstall the system version of gyp.
+
+On Fedora you would do the following:
+
+``` command-line
+$ sudo yum remove gyp
+```
+
+###### Linux build error reports
+
+Use [this search](https://github.com/atom/atom/search?q=label%3Abuild-error+label%3Alinux&type=Issues)
+to get a list of reports about build errors on Linux.
+
+{{/linux}}
+
+#### Running in Development Mode
+
+Once you have a local copy of Atom cloned and built, you can then run Atom in Development Mode. But first, you have to set the `ATOM_DEV_RESOURCE_PATH` environment variable. It defaults to <span class="platform-mac platform-linux">`~/github`</span><span class="platform-windows">`%USERPROFILE%\github`</span> which then expects the `atom/atom` repository to be cloned into <span class="platform-mac platform-linux">`$ATOM_DEV_RESOURCE_PATH/atom`</span><span class="platform-windows">`%ATOM_DEV_RESOURCE_PATH%\atom`</span>. To run Atom in Dev Mode, use the `--dev` parameter from the terminal:
+
+``` command-line
+$ atom --dev <em>path-to-open</em>
+```
+
+There are a couple benefits of running Atom in Dev Mode:
+
+1. When the `ATOM_DEV_RESOURCE_PATH` environment variable is set correctly, Atom is run using the source code from your local `atom/atom` repository. This means that you don't have to run `script/build` every time you change code. Just restart Atom 👍
+1. Packages that exist in <span class="platform-mac platform-linux">`~/.atom/dev/packages`</span><span class="platform-windows">`%USERPROFILE%\.atom\dev\packages`</span> are loaded instead of packages of the same name normally loaded from other locations. This means that you can have development versions of packages you use loaded but easily go back to the stable versions by launching without Dev Mode.
+
+#### Running Atom Core Tests Locally
+
+In order to run Atom Core tests from the terminal:
```

Comment:
The ATOM_DEV resource path needs to be setup correctly at least when using the GUI runner which is not mentioned.

Reply:
I don't know how atom --test works since I never used it

Reply:
I think we're pushing people more toward using `atom --test` so that's why I didn't mention the GUI method. I'll add a mention of the `ATOM_DEV_RESOURCE_PATH` thing though :+1: